railties/test/isolation/abstract_unit.rb

```diff
@@ -501,12 +501,18 @@ class ActiveSupport::TestCase
   unless Dir.exist?("#{assets_path}/node_modules")
     Dir.chdir(assets_path) { `yarn install` }
   end
-  FileUtils.cp("#{assets_path}/package.json", "#{app_template_path}/package.json")
   FileUtils.cp("#{assets_path}/config/webpacker.yml", "#{app_template_path}/config/webpacker.yml")
   FileUtils.cp_r("#{assets_path}/config/webpack", "#{app_template_path}/config/webpack")
   FileUtils.ln_s("#{assets_path}/node_modules", "#{app_template_path}/node_modules")
   FileUtils.chdir(app_template_path) { `bin/rails webpacker:binstubs` }
 
+  # package.json in assets_path contains relative paths. Changing them to
+  # absolute and running `yarn install` to generate matching yarn.lock.
+  contents = File.read("#{assets_path}/package.json")
+  contents.gsub!(/file:\W*/, "file:#{RAILS_FRAMEWORK_ROOT}/")
+  File.write("#{app_template_path}/package.json", contents)
+  Dir.chdir(app_template_path) { `yarn install` }
```

Comment:
I want to avoid run `yarn install` inside the test.

The fact that the tests passed in the original condition mean that the JS library is unnecessary for testing.
So I think better to generate `package.json` that library specification is empty in the setup of `assets_test.rb`

Reply:
i’m not sure it’s unnecessary, all the needed packages were already there - it’s just the package.json was wrong, so yarn install failed but something might be requiring modules and succeeding

I’ll use an empty package.json and see if anything fails